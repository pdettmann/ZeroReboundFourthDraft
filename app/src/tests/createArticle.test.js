import React from 'react';
import CreateArticle from '../createArticle';
import renderer from 'react-test-renderer';
import { render, unmountComponentAtNode } from "react-dom";
import { UserProvider } from '../userContext';
import { act } from "react-dom/test-utils";
import { BrowserRouter as Router } from 'react-router-dom';
import {  fireEvent } from "@testing-library/react";

let container = null;
beforeEach(() => {
	container = document.createElement("div");
	document.body.appendChild(container);
});

afterEach(() => {
	unmountComponentAtNode(container);
	container.remove();
	container = null;
});

test('CreateArticle renders correctly', () => {
		const component = renderer.create(
				<UserProvider>
						<CreateArticle />
				</UserProvider>
			);
			let tree = component.toJSON();
			expect(tree).toMatchSnapshot();
});

test('Article gets created', async () => {
	const mockApiClient = {
		post: ((path) => {
			if (path === '/article/create') {
				return Promise.resolve({
					data: {
						article: {
							title: 'something',
							text: 'something else',
						},
				 }
				});
			}
			return Promise.reject();
		}),
	};

	await act (async () => {
		render(
			<UserProvider>
					<Router>
						<CreateArticle apiClient={mockApiClient} />
					</Router>
			</UserProvider>,
			container
		);
	});

	await act(async () => {
		const button =  container.querySelector('#createArticle');
		fireEvent.click(button);
	});
// add test
});
