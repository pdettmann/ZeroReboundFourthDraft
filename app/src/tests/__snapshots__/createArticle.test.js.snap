// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateArticle renders correctly 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Home
  </button>
  <button
    onClick={[Function]}
  >
    Profile
  </button>
  <h1>
    Create an Article
  </h1>
  <input
    className="titleCreateArticle"
    onChange={[Function]}
    placeholder="Title"
    type="text"
  />
  <br />
  <textarea
    className="textCreateArticle"
    onChange={[Function]}
    placeholder="Copy or write your article text here..."
    type="text"
  />
  <br />
  <button
    id="createArticle"
    onClick={[Function]}
    type="submit"
  >
     Create 
  </button>
  <br />
</div>
`;
